core/node/eth_watch/src/event_processors/priority_ops.rs

```diff
@@ -70,33 +71,48 @@ impl EventProcessor for PriorityOpsEventProcessor {
             .into_iter()
             .skip_while(|tx| tx.serial_id() < self.next_expected_priority_id)
             .collect();
-        let (Some(first_new), Some(last_new)) = (new_ops.first(), new_ops.last()) else {
-            return Ok(());
+        let Some(first_new) = new_ops.first() else {
+            return Ok(events_count);
         };
         assert_eq!(
             first_new.serial_id(),
             self.next_expected_priority_id,
             "priority transaction serial id mismatch"
         );
-        let next_expected_priority_id = last_new.serial_id().next();
+        let mut next_expected_priority_id = self.next_expected_priority_id;
 
         let stage_latency = METRICS.poll_eth_node[&PollStage::PersistL1Txs].start();
         APP_METRICS.processed_txs[&TxStage::added_to_mempool()].inc();
         APP_METRICS.processed_l1_txs[&TxStage::added_to_mempool()].inc();
+        let processed_priority_transactions = client.get_total_priority_txs().await?;
+        let mut processed_events_count = 0;
         for new_op in new_ops {
+            if processed_priority_transactions <= new_op.serial_id().0 {
+                break;
+            }
```

Comment:
can we avoid using `mut`, `break` and e.g. do `take_while` instead? I think it will look nicer

Reply:
also `processed_priority_transactions` seems to be off here as it doesn't account for `skip_while` executed before.

Reply:
nice catch, I've also refactored this part of the code into more functional as you suggested, it looks much nicer now